### PR TITLE
Move the Classic block to the packages

### PR DESCRIPTION
**Pull request by youknowriad**
In order to merge Gutenberg into Core easily, all the Gutenberg JavaScript Core must be shared a reusable npm package. This PR moves the freeform block to the packages folder.

The idea is to test whether the `wp.oldEditor` global (provided by WP) is available or not before registering the block. An alternative proposed by @gziolo is to move this block to `edit-post` module. I think both alternatives are good options. We'll probably have to revisit at some point either way so I just went with the simplest one for noow.

This also means the block library is fully available as a package. (so a bunch of files removed etc...)

**Testing instructions**

Make sure you can use the classic block properly.

### CODE REVIEW

**Comment by iseulde** [comment]
Should the folders be renamed to `classic` instead of `freeform`?

**Comment by tofumatt** [comment]
Yes; the "freeform" name is quite confusing! 😅

**Comment by youknowriad** [comment]
@tofumatt @iseulde The reason it's kept as "classic" is because we can't change the name of the block at the moment. I can rename the folder to "classic" but it feels like we'd create an inconsistency between block names and folders which all other blocks follow.

**Comment by gziolo** [comment]
> The reason it's kept as "classic" is because we can't change the name of the block at the moment. I can rename the folder to "classic" but it feels like we'd create an inconsistency between block names and folders which all other blocks follow.

It's called Classic in the UI:

screen shot 2018-10-08 at 17 12 19

I don't like exceptions, but in this case, it is only an internal name. We refer to it as classic every time we chat. I would vote to rename the folder and the variable assigned with import and leave the rest as is.

**Review by gziolo — APPROVED** [review]
This looks great. We could discuss whether it should land in `edit-post` but this makes all the things easier for the time being and removes a lot of dead code. I would still consider renaming the folder from `freeform` to `classic`.

**Comment by youknowriad** [comment]
I'll wait for a review of #10396 to merge this one as it's targetting that branch.

File: packages/block-library/src/editor.scss [hunk 1]
```diff
@@ -7,6 +7,7 @@
 @import "./cover-image/editor.scss";
 @import "./embed/editor.scss";
 @import "./file/editor.scss";
+@import "./freeform/editor.scss";
```
Comment (gziolo) [inline packages/block-library/src/editor.scss]: this should be updated, too

File: packages/block-library/src/index.js [hunk 2]
```diff
@@ -85,9 +90,16 @@ export const registerCoreBlocks = () => {
...
+	].forEach( ( block ) => {
+		if ( ! block ) {
+			return;
+		}
+		const { name, settings } = block;
 		registerBlockType( name, settings );
 	} );
 
 	setDefaultBlockName( paragraph.name );
+	if ( window.wp && window.wp.oldEditor ) {
```
[the remark this answers is not in the document]
Comment (gziolo) [reply packages/block-library/src/index.js]: I don't think it is important in this case, but I'm fine with leaving it as it is.

**Comment by talldan** [comment]
Hey @youknowriad - I’m seeing a warning in latest master:
```
Warning: filemtime(): stat failed for /var/www/html/wp-content/plugins/gutenberg/build/block-library/edit-blocks.css in /var/www/html/wp-content/plugins/gutenberg/lib/client-assets.php on line 741
```
It's still there after I cleaned the repo using `git clean -dfx`. I noticed some references to edit-blocks.css in this file, so I imagine there's a good chance it's related.

**Comment by gziolo** [comment]
@talldan, looking into it.

**Comment by gziolo** [comment]
I can't reproduce:

screen shot 2018-10-10 at 10 43 31

All 3 styles for `block-library` load properly. Also PHP code doesn't reference the file you shared:

```php
wp_register_style(
		'wp-edit-blocks',
		gutenberg_url( 'build/block-library/editor.css' ),
		array(
			'wp-components',
			'wp-editor',
			// Always include visual styles so the editor never appears broken.
			'wp-block-library-theme',
		),
		filemtime( gutenberg_dir_path() . 'build/block-library/editor.css' )
	);
```

**Comment by talldan** [comment]
Hmm, unusual. Restarted my docker env and it's fine now. Thanks for looking into it.

**Pull request merged**